### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by babakfp** [pocketbase/pocketbase#1741] [Q&A] [answered]
Hey
I'm new to Pocketbase and DB stuff, so is there a reason that these things are done this way?

1. Why it's not possible to edit the password field from the default users collection?
2. Why can't I remove email related stuff from the default users collection?

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
1. Users can change their password if they also provide the `oldPassword` field. Auth records that satisfies the auth collection "Manage API" rule can change everything (including the `verified` state), similar to the admins.

2. Because there are various parts of the application that relies on the email field (eg. OAuth2 linking, password reset, verification, etc.) and making it optional will just complicate the API, documentations and testing due to the required conditional handling and for now, at least for me, it is easier to always have this field by default.

**Reply by babakfp** [reply]
1. Sorry, I mean from the Admin UI (without needing the old password).
2. 👍

**Reply by ganigeorgiev** [reply]
About 1) - I'm not sure what do you mean. You can change the user password by clicking on the user > and enable the "Change password" toggle (admins don't have to enter the old password). If you are asking to remove the password field entirely, then this is not possible for similar reasons as the email field.

**Reply by babakfp** [reply]
![Screenshot 2023-02-01 121148](https://user-images.githubusercontent.com/44144724/216152800-0a0efdc5-07e3-4d86-a2d7-889c4ca84b2b.png)

I can sware it wasn't there 5 minutes ago😅 (sorry, my bad🌷)

Thank you🧡
